org-roam.el

```diff
@@ -382,12 +382,14 @@ current buffer is used."
 
 (defun org-roam--extract-ref ()
   "Extract the ref from current buffer and return the type and the key of the ref."
-  (if-let ((ref (cdr (assoc "ROAM_KEY" (org-roam--extract-global-props '("ROAM_KEY"))))))
-      (let* ((type (org-roam--ref-type ref))
-             (key (cond ((string= "cite" type)
-                         (s-chop-prefix (org-roam--cite-prefix ref) ref))
-                        (t ref))))
-        (cons type key))
+  (if-let* ((ref (cdr (assoc "ROAM_KEY"
+                             (org-roam--extract-global-props '("ROAM_KEY")))))
+            (not-empty (not (string-empty-p ref)))
```

Comment:
I don't think it's particularly good style to define a variable in an `if-let` form which is just meant to short-circuit the condition.  I could probably rewrite it with a `catch` and a thread (`->`), but I don't think it's worth it.

Reply:
I think the correct approach is to not have the `#+ROAM_KEY` property at all in the Org file.

Reply:
I thought the  same.  Maybe we could just `user-error` on the empty-string, then?

Reply:
yeah, that sounds like a better approach. Could probably add that as one of the checks later on.

Reply:
Or catch the `peculiar error` on inserting the ref, Saying that the ref `""` already exists is quite telling.

Reply:
Could work.  I'll try things out when I'm done eating.